core/src/main/scala/org/apache/spark/status/api/v1/ApiRootResource.scala

```diff
@@ -184,14 +184,27 @@ private[v1] class ApiRootResource extends ApiRequestContext {
   @Path("applications/{appId}/logs")
   def getEventLogs(
       @PathParam("appId") appId: String): EventLogDownloadResource = {
-    new EventLogDownloadResource(uiRoot, appId, None)
+    try {
+      // withSparkUI will throw NotFoundException if attemptId is existed for this application.
+      // So we need to try again with attempt id "1".
+      withSparkUI(appId, None) { _ =>
```

Comment:
Hmm... optimally you wouldn't do this, because loading the UI can be expensive. That would require recording extra information when listing applications and some massaging of internal APIs, so this is probably OK for now, but would be nice to track it separately as an enhancement.